### ISSUE TITLE
Maze Generation is too sparse

### ISSUE THREAD

**Issue by d1llonshu** [Pingry-Java/MazeSolver#18]
Generates a maze with lots of passable blocks and not very many walls.  The issue probably is not regarding our algorithm, more likely the implementation.

**Comment by jorndorff** [comment]
I agree. I tried to debug but didn't get very far. Here are a few things I noticed.

1. You have two different maze squares at location 1,1. One is the anonymous square you passed to super and another is made when the rest of the grid is filled.

2. No documentation anywhere. I opened a separate issue for that at #19 .

3. Definitely not the bug, but your class is named with a lowercase letter.

**Comment by jorndorff** [comment]
For example, what does `eligible` do, and what is the difference between the following Lists in that method, `neighbors`, `eligibleNeighbors`, `visitedNeighbors`

**Comment by jorndorff** [comment]
I want to help with this, but I don't know how to run the text-mode program you were using to test the maze. Where is that code?

**Comment by drew-beckmen** [comment]
We have a print maze function - should be there in generate maze. can call it on an instance of generate maze

**Comment by jorndorff** [comment]
I just pushed a branch called debuggingGenerateMaze that has some cleaned up and thoroughly commented code as well as a main programming with good debugging output. Check it out and you'll see the problem.

Bottom Line: We defined eligible nodes incorrectly. Once we figure out what really makes a node eligible, we'll be in business.

**Comment by jorndorff** [comment]
Incorrect:
`If a node has two UNVISITED neighbors, it is ELIGIBLE`
`if (visitedNeighbors.size() >= 2)`

Correct:
`If a node two VISITED neighbors, it is INELIGIBLE`
`if(visitedNeighbors.size() > 2)`

Fixing this will likely expose a few other bugs that I've opened issues for pre-emptively.

Don't forget to check out my branch `debuggingGenerateMaze`
